rust/agama-cli/src/lib.rs

```diff
@@ -217,29 +217,39 @@ pub fn download_file(url: &str, path: &PathBuf) -> Result<(), ServiceError> {
     Ok(())
 }
 
-pub async fn run_command(cli: Cli) -> Result<(), ServiceError> {
-    // somehow check whether we need to ask user for self-signed certificate acceptance
-    let api_url = cli.opts.api.trim_end_matches('/').to_string();
-
+async fn build_http_client(
+    api_url: &str,
+    insecure: bool,
+    authenticated: bool,
+) -> Result<BaseHTTPClient, ServiceError> {
     let mut client = BaseHTTPClient::default();
 
-    client.base_url = api_url.clone();
+    client.base_url = api_url.to_string();
 
-    if allowed_insecure_api(cli.opts.insecure, api_url.clone()).await? {
+    if allowed_insecure_api(insecure, api_url.to_string()).await? {
```

Comment:
Question: why not using client.base_url here?

Reply:
For no particular reason. I just moved the code to a function with the minimal changes. I have updated that call.